index.js

```diff
@@ -38,6 +38,9 @@ function AllureReporter(runner, opts) {
     });
 
     runner.on("fail", function(test, err) {
+        if(!allureReporter.getCurrentTest()) {
```

Comment:
I agree that it doesn't fix the core of the problem. However, It will
protect you against cases where there is no test case registered yet and
you attempt to end a test (like in a before hook) fairly gracefully. It
looked like most of the logic is actually in the allure-common library you
are using and I did not have the time to dive into that.

On Mon, Feb 29, 2016 at 4:38 AM, Boris Serdiuk <notifications@github.com>
wrote:

> In index.js
> <https://github.com/allure-framework/mocha-allure-reporter/pull/12#discussion_r54386663>
> :
>
> > @@ -38,6 +38,9 @@ function AllureReporter(runner, opts) {
> >      });
> >
> >      runner.on("fail", function(test, err) {
> > +        if(!allureReporter.getCurrentTest()) {
>
> I am not quite sure, that it should be checked in that way, but anyway
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/allure-framework/mocha-allure-reporter/pull/12/files#r54386663>
> .
>